x-pack/plugins/security_solution/server/endpoint/services/agent/clients/lib/base_agent_status_client.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/logging';
+import type { AgentStatusRecords } from '../../../../../../common/endpoint/types/agents';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import type { EndpointAppContextService } from '../../../../endpoint_app_context_services';
+import type { AgentStatusClientInterface } from './types';
+import { AgentStatusClientError, AgentStatusNotSupportedError } from '../errors';
+import { HOST_NOT_ENROLLED } from '../../../actions/clients/lib/base_response_actions_client';
+
+export interface AgentStatusClientOptions {
+  endpointService: EndpointAppContextService;
+  esClient: ElasticsearchClient;
+}
+
+export type AgentStatusClientValidateRequestResponse =
+  | {
+      isValid: true;
+      error: undefined;
+    }
+  | {
+      isValid: false;
+      error: AgentStatusClientError;
+    };
+
+export abstract class AgentStatusClient implements AgentStatusClientInterface {
+  protected readonly log: Logger;
+  protected abstract readonly agentType: ResponseActionAgentType;
+
+  constructor(protected readonly options: AgentStatusClientOptions) {
+    this.log = options.endpointService.createLogger(this.constructor.name ?? 'AgentStatusClient');
+  }
+
+  /**
+   * Provides validations against a response action request and returns the result.
+   * Checks made should be generic to all response actions and not specific to any one action.
+   *
+   * @param agentIds
+   * @param agentType
+   * @protected
+   */
+  protected async validateRequest(
+    agentIds: string[],
+    agentType: ResponseActionAgentType
```

Comment:
why does this method accept `agentType`? Since `agentType` is required to be defined in the class and the method is `protected`, I can't see why its needed.